crates/vm/src/system/memory/controller/dimensions.rs

```diff
@@ -17,6 +17,14 @@ pub struct MemoryDimensions {
 }
 
 impl MemoryDimensions {
+    pub fn from_config(config: &MemoryConfig) -> Self {
+        Self {
+            as_height: config.as_height,
+            address_height: config.pointer_max_bits - log2_strict_usize(CHUNK),
```

Comment:
is this correct? is `MemoryDimensions` just a property of the tree? if so, maybe we should move it to `tree.rs`